### DISCUSSION TITLE
Problems with connecting srsUE and srsRAN over the air and using SMA

### DISCUSSION THREAD

**Discussion by HesselTjeerdsma** [srsran/srsRAN_Project#137] [General Help]
Hi,

I am already stuck for quite some time with connecting srsUE and srsRAN over the air or over a SMA cable (with -30db attenuator). The setup has been set up like [here](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html). Next to that, I am using a USRP x310 for the srsRAN side and USRP B210 for srsUE side. I have tested both an over the air setup with antennas and a direct connection using sma cables. Also, the core network works as the setup with ZeroMQ works without issues. 

In the debug.zip are the config, logs and pcap files of the gnb and ue.

Could somebody please help me! Really looking forward to getting this setup to work. Thank you!

[debug.zip](https://github.com/srsran/srsRAN_Project/files/12122540/debug.zip)

**Comment by andrepuschmann** [comment]
The UE doesn't find any cell. There is something wrong with the RF setup, frequencies, gains, antenna.

**Reply by HesselTjeerdsma** [reply]
Could you give me a small hint what would be good settings for the gains? Currently, I am connected directly over SMA cables with a -30db attentuator in between. The ue (usrp b210) is configured with a tx and rx gain of 10, the gnb (usrp x310) is configured with tx and rx gain of 15.

**Reply by ismagom** [reply]
with 30 dB attenuator, you can increase the tx_gain in the B210 to maybe 50-60 dB, and the rx_gain to 40. The X310 tx_gain is fine at 15-20 and rx_gain to 15.

**Reply by HesselTjeerdsma** [reply]
Yes, thank you. That setup now works and the UE finds the cell. However, now the log file is full SYNC: detected out-of-sync (log files are included). What can I do about that? I am trying to use the GPSDO but I cannot seem to get a gps lock on the x310. Any advice for fixing this setup? Thank you!

[ue.log](https://github.com/srsran/srsRAN_Project/files/12172695/ue.log)

**Reply by HesselTjeerdsma** [reply]
Switching to a different x310 helped with the GPS problem. Now, the X310 has a gps lock. However, the srsUE is still not connecting. I have attached to logs of the ue again. Thank you! 
[ue.log](https://github.com/srsran/srsRAN_Project/files/12173054/ue.log)

**Reply by HesselTjeerdsma** [reply]
Setting the srate to 11.52 in the gnb helped as well. The UE seems to get further in the process. But again, it cannot attach fully yet. I have attached the ue.logs and the gnb_ngap.pcap again. Somebody please can give me a hint for a next step? Thank you! 
[ue.log](https://github.com/srsran/srsRAN_Project/files/12173895/ue.log) 
[gnb_ngap.pcap.zip](https://github.com/srsran/srsRAN_Project/files/12173920/gnb_ngap.pcap.zip)

**Reply by dhiaboujebha** [reply]
@HesselTjeerdsma Have you managed to solve the issue? I have the same problem, could you please share your configuration files if it is working? Thank you in advance.

**Reply by HesselTjeerdsma** [reply]
@dhiaboujebha No, sorry. We switched to OpenAirInterface now...

**Reply by andrepuschmann** [reply]
Hey guys - please excuse the long delay in getting back here.

@HesselTjeerdsma Just looking at the UE logs you shared it seems the UE is trying to PRACH, gets the RAR but then Msg3 fails. You need to share the gNB logs to verify but my first guess is that there is still an issue with UL gain settings, etc and the gNB simply fails to decode Msg3.

The UE tries to attach multiple times but then simply stays there and doesn't try again - that's expected though. But it's RF setup issue for sure.

If you've tried the OAI UE and it worked better you've might be using "better" RF settings for your particular scenario.

Hope that helps.

**Reply by dhiaboujebha** [reply]
@andrepuschmann could you please take a look at the issue https://github.com/srsran/srsRAN_Project/issues/199 . I have added my logs there.

**Comment by pgawlowicz** [comment]
@HesselTjeerdsma It seems that your GNB is configured to run with Bandwidth = 20MHz, while UE with 10MHz. Due to the srsUE limitations, this parameter has to be consistent between gnb and srsUE. Also, note that back in September, the srsUE operated only with 10MHz, with the new srsRAN_4G release (23.11), it can operate with 5/10/15/20MHz. But still BW has to be the same in gnb and srsUE.
